File: 219.contains-duplicate-ii.py
#
# @lc app=leetcode id=219 lang=python3
#
# [219] Contains Duplicate II
#

# @lc code=start
from collections import Counter
class Solution:
    def containsNearbyDuplicate(self, nums: list[int], k: int) -> bool:
        hashmap = Counter()
        ans = 0
        for i, num in enumerate(nums):
            if num in hashmap and i - hashmap[num] <= k:
                return True
            hashmap[num] = i
        return False     
# @lc code=end

File: 219.contains-duplicate-ii.java
/*
 * @lc app=leetcode id=219 lang=java
 *
 * [219] Contains Duplicate II
 */

// @lc code=start
import java.util.HashMap;
import java.util.Map;
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int minDis = Integer.MAX_VALUE;
        for(int i = 0; i < nums.length;i++){
            if(map.containsKey(nums[i])){
                minDis = Math.min(minDis, i - map.get(nums[i]));
                map.put(nums[i], i);
            }
            else{
                map.put(nums[i], i);
            }
            if (minDis <= k){
                return true;
            }
        }
        return false; 
    }
}
// @lc code=end

